Snake/BusinessLayer/Engine.cs

```diff
@@ -141,13 +151,48 @@ public Engine(gameMode mode , int mazeMode, int length = mazeRenderLength, int w
                     bool isAIValid = true;
                     int previousX = newAI.XCoordinate;
                     int previousY = newAI.YCoordinate;
+
                     do
                     {
                         newAI.MoveAI(previousX, previousY);
+
                         isAIValid = validateNewAILocation(newAI);
+
                     } while (!isAIValid);
+
                     mazeArray[newAI.XCoordinate, newAI.YCoordinate] = Elements.AI;
 
+                    previousX = newSmartAI.XCoordinate;
+                    previousY = newSmartAI.YCoordinate;
+                    mazeArray[newSmartAI.XCoordinate, newSmartAI.YCoordinate] = 0;
+
+                    int temp = 0;
+                    Random rnd = new Random();
+                   
+                    do
+                    {
+                        if (smartMove)
+                        {
+                            temp = rnd.Next(2);
+                            newSmartAI.SmartMove(temp, SnakeCurrentPosition, previousX, previousY);
+                            isAIValid = validateNewAILocation(newSmartAI);
+                            if(isAIValid)
+                            {
+                                smartMove = false;
+                            }
+                            else
+                            {
+                                newSmartAI.XCoordinate = previousX;

```

Comment:
Looks like we need to keep track of the previous position in the AI - this would allow us to remove the setters again!